repos/spack_repo/builtin/packages/py_geopmdpy/package.py

```diff
@@ -13,28 +13,43 @@ class PyGeopmdpy(PythonPackage):
 
     homepage = "https://geopm.github.io"
     git = "https://github.com/geopm/geopm.git"
-    url = "https://github.com/geopm/geopm/tarball/v3.1.0"
+    url = "https://github.com/geopm/geopm/tarball/v3.2.0"
 
     maintainers("bgeltz", "cmcantalupo")
     license("BSD-3-Clause")
     tags = ["e4s"]
 
+    variant("grpc", default=False, when="@3.2:", description="Enable gRPC support")
+
     version("develop", branch="dev", get_full_repo=True)
+    version("3.2.0", sha256="b708233e1bfda66408c500f2ac0cbaf042140870bffdced12dd7cabbd18e0025")
     version("3.1.0", sha256="2d890cad906fd2008dc57f4e06537695d4a027e1dc1ed92feed4d81bb1a1449e")
-    version("3.0.1", sha256="32ba1948de58815ee055470dcdea64593d1113a6cad70ce00ab0286c127f8234")
+    version(
+        "3.0.1",
+        sha256="32ba1948de58815ee055470dcdea64593d1113a6cad70ce00ab0286c127f8234",
+        deprecated=True
+    )
 
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
     depends_on("fortran", type="build")  # generated
 
+    for ver in ["3.1.0", "3.2.0", "develop"]:
+        depends_on(f"geopm-service@{ver}", type=("build", "test", "run"), when=f"@{ver}")
     depends_on("py-dasbus@1.6.0:", type=("build", "run"))
-    depends_on("py-cffi@1.14.5:", type="run")
-    depends_on("py-psutil@5.8.0:", type="run")
-    depends_on("py-jsonschema@3.2.0:", type="run")
+    depends_on("py-cffi@1.14.5:", when="@3.0.1:3.1", type="run")
+    depends_on("py-cffi@1.14.5:", when="@3.2:", type=("build", "run"))
+    depends_on("py-psutil@5.8.0:", when="@3.0.1:3.1", type="run")
+    depends_on("py-psutil@5.8.0:", when="@3.2:", type=("test", "run"))
+    depends_on("py-jsonschema@3.2.0:", when="@3.0.1:3.1", type="run")
+    depends_on("py-jsonschema@3.2.0:", when="@3.2:", type=("test", "run"))
     depends_on("py-pyyaml@6.0:", type="run")
```

Comment:
Is this supposed to always be available or is/would it be tied to a `stats` variant, per `3.2.0`'s `pyproject.toml`:

```
stats = [
    "seaborn>=0.11.2",
    "pyyaml>=6.0.0",
]
```

If always, would `py-seaborn@0.11.2:` also be required?